Error using WebSockets on Heroku

##### **Overview of the issue**

When I deploy the app to Heroku and load the generated "user tracking" page that uses websockets the following error appears on logs and the page don't show the tracking data:

>Caused by: org.springframework.web.socket.sockjs.SockJsTransportFailureException: WebSocket handshake failure; nested exception is java.lang.IllegalArgumentException: >No 'javax.websocket.server.ServerContainer' ServletContext attribute. Are you running in a Servlet container that supports JSR-356?

##### **Motivation for or Use Case**

I want to use WebSockets without errors on console/app.

##### **Reproduce the error**

1. Create a monolith angular 1.x app with websockets
2. Deploy to Heroku with the command (found on gitlab-ci.yml generated by JHipster):
`./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy-war -DskipTests -Pprod -Dheroku.appName=myapp`
3. Navigate to the user tracking page (Ex.: https://myapp.herokuapp.com/#/tracker)
4. Open another browser, log in and open any page on the app
5. See Heroku logs for the error


##### **Related issues**

None that I have found.

##### **Suggest a Fix**

This error appears to happen because the command deploy-war of heroku-maven-plugin uses webapp-runner.jar to run the war on Heroku, but since "myapp" is a SpringBoot one we don't need this webapp-runner.jar.

If we use the command **deploy instead of deploy-war** the app works without errors (and with a bit small deploy):
`./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy -DskipTests -Pprod -Dheroku.appName=myapp`

##### **JHipster Version(s)**

```
myapp@0.0.0 C:\myapp\server-app
`-- generator-jhipster@4.0.2

```

##### **JHipster configuration**

```yaml
{
  "generator-jhipster": {
    "jhipsterVersion": "4.0.2",
    "baseName": "myapp",
    "packageName": "com.myapp",
    "packageFolder": "com/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "hibernateCache": "hazelcast",
    "clusteredHttpSession": false,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "jwtSecretKey": "abcd",
    "useSass": false,
    "clientPackageManager": "npm",
    "applicationType": "monolith",
    "clientFramework": "angular1",
    "testFrameworks": [
      "gatling",
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "pt-br",
    "languages": [
      "pt-br"
    ],
    "herokuAppName": "myapp"
  }
}
```


##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

TipoDocumento.json

```yaml
{
    "fluentMethods": true,
    "relationships": [],
    "fields": [
        {
            "fieldName": "nome",
            "fieldType": "String",
            "fieldValidateRules": [
                "required",
                "maxlength"
            ],
            "fieldValidateRulesMaxlength": "30"
        }
    ],
    "changelogDate": "20170301192641",
    "dto": "no",
    "service": "no",
    "entityTableName": "tipo_documento",
    "pagination": "no"
}

```

TipoEndereco.json

```yaml
{
    "fluentMethods": true,
    "relationships": [],
    "fields": [
        {
            "fieldName": "nome",
            "fieldType": "String",
            "fieldValidateRules": [
                "required",
                "maxlength"
            ],
            "fieldValidateRulesMaxlength": 20
        }
    ],
    "changelogDate": "20170228171654",
    "dto": "no",
    "service": "no",
    "entityTableName": "tipo_endereco",
    "pagination": "no"
}

```

##### **Browsers and Operating System**

All browsers.

java version "1.8.0_91"
Java(TM) SE Runtime Environment (build 1.8.0_91-b15)
Java HotSpot(TM) Client VM (build 25.91-b15, mixed mode)

git version 1.9.5.msysgit.0

node: v6.9.1

npm: 3.10.8

bower: 1.8.0

gulp:
[02:59:27] CLI version 3.9.1
[02:59:27] Local version 3.9.1

- [ x ] Checking this box is mandatory (this is just to show you read everything)

--- a/generators/ci-cd/templates/_.gitlab-ci.yml
+++ b/generators/ci-cd/templates/_.gitlab-ci.yml
@@ -169,7 +169,7 @@ gatling-test:
 maven-deploy:
     stage: deploy
      script:
-        - ./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy-war -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>
+        - ./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>
     <%_ } else { _%>
 maven-package:
     stage: package

--- a/generators/ci-cd/templates/_circle.yml
+++ b/generators/ci-cd/templates/_circle.yml
@@ -63,7 +63,7 @@ test:
 <%_ } _%>
 <%_ if (buildTool === 'maven') { _%>
     <%_ if (heroku.includes('circle')) { _%>
-        - ./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy-war -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>
+        - ./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>
     <%_ } else { _%>
         - ./mvnw package -Pprod -DskipTests
     <%_ } _%>

--- a/generators/ci-cd/templates/_travis.yml
+++ b/generators/ci-cd/templates/_travis.yml
@@ -79,7 +79,7 @@ script:
 <%_ } _%>
 <%_ if (buildTool === 'maven') { _%>
   <%_ if (heroku.includes('travis')) { _%>
-  - ./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy-war -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>
+  - ./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>
   <%_ } else { _%>
   - ./mvnw package -Pprod -DskipTests
   <%_ } _%>

--- a/generators/ci-cd/templates/jenkins/_Jenkinsfile
+++ b/generators/ci-cd/templates/jenkins/_Jenkinsfile
@@ -74,7 +74,7 @@ node {
     <%_ } _%>
     <%_ if (heroku.includes('jenkins')) { _%>
 <%= indent %>    stage('package and deploy') {
-<%= indent %>        sh "./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy-war -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>"
+<%= indent %>        sh "./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>"
     <%_ } else { _%>
 <%= indent %>    stage('packaging') {
 <%= indent %>        sh "./mvnw package -Pprod -DskipTests"

